tests/components/zeroconf/test_init.py

```diff
@@ -135,16 +158,62 @@ async def test_homekit_match_full(hass, mock_zeroconf):
     ), patch.object(
         hass.config_entries.flow, "async_init"
     ) as mock_config_flow, patch.object(
-        zeroconf, "ServiceBrowser", side_effect=service_update_mock
+        zeroconf, "HaServiceBrowser", side_effect=service_update_mock
     ) as mock_service_browser:
-        mock_zeroconf.get_service_info.side_effect = get_homekit_info_mock("BSB002")
+        mock_zeroconf.get_service_info.side_effect = get_homekit_info_mock(
+            "BSB002", HOMEKIT_STATUS_UNPAIRED
+        )
         assert await async_setup_component(hass, zeroconf.DOMAIN, {zeroconf.DOMAIN: {}})
 
+    homekit_mock = get_homekit_info_mock("BSB002", HOMEKIT_STATUS_UNPAIRED)
+    info = homekit_mock("_hap._tcp.local.", "BSB002._hap._tcp.local.")
+    import pprint
+
+    pprint.pprint(["homekit", info])
     assert len(mock_service_browser.mock_calls) == 1
     assert len(mock_config_flow.mock_calls) == 1
     assert mock_config_flow.mock_calls[0][1][0] == "hue"
 
 
+async def test_homekit_already_paired(hass, mock_zeroconf):
+    """Test that an already paired device is sent to homekit_controller."""
+    with patch.dict(
+        zc_gen.ZEROCONF, {zeroconf.HOMEKIT_TYPE: ["homekit_controller"]}, clear=True
+    ), patch.object(
+        hass.config_entries.flow, "async_init"
+    ) as mock_config_flow, patch.object(
+        zeroconf, "HaServiceBrowser", side_effect=service_update_mock
+    ) as mock_service_browser:
+        mock_zeroconf.get_service_info.side_effect = get_homekit_info_mock(
+            "tado", HOMEKIT_STATUS_PAIRED
+        )
+        assert await async_setup_component(hass, zeroconf.DOMAIN, {zeroconf.DOMAIN: {}})
+
+    assert len(mock_service_browser.mock_calls) == 1
+    assert len(mock_config_flow.mock_calls) == 2
+    assert mock_config_flow.mock_calls[0][1][0] == "tado"
+    assert mock_config_flow.mock_calls[1][1][0] == "homekit_controller"
+
+
+async def test_homekit_invalid_paring_status(hass, mock_zeroconf):
+    """Test that missing paring data is not sent to homekit_controller."""
+    with patch.dict(
+        zc_gen.ZEROCONF, {zeroconf.HOMEKIT_TYPE: ["homekit_controller"]}, clear=True
+    ), patch.object(
+        hass.config_entries.flow, "async_init"
+    ) as mock_config_flow, patch.object(
+        zeroconf, "HaServiceBrowser", side_effect=service_update_mock
+    ) as mock_service_browser:
+        mock_zeroconf.get_service_info.side_effect = get_homekit_info_mock(
+            "tado", b"invalid"
+        )
+        assert await async_setup_component(hass, zeroconf.DOMAIN, {zeroconf.DOMAIN: {}})
+
+    assert len(mock_service_browser.mock_calls) == 1
+    assert len(mock_config_flow.mock_calls) == 1
+    assert mock_config_flow.mock_calls[0][1][0] == "tado"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)